src/schema/v2/discoverArtworks.ts

```diff
@@ -0,0 +1,44 @@
+import { GraphQLString, GraphQLFieldConfig } from "graphql"
+import { ResolverContext } from "types/graphql"
+import { artworkConnection } from "./artwork"
+import { connectionFromArray } from "graphql-relay"
+import { pageable } from "relay-cursor-paging"
+import gql from "lib/gql"
+
+export const DiscoverArtworks: GraphQLFieldConfig<void, ResolverContext> = {
+  type: artworkConnection.connectionType,
+  args: pageable({
+    userId: { type: GraphQLString },
+  }),
+  resolve: async (_root, args, { weaviateGraphqlLoader, artworksLoader }) => {
+    if (!weaviateGraphqlLoader) return
+    const { userId } = args
+
+    const query = gql`
+      query discoveryArtworksQuery {
+        {
```

Comment:
Think we need to remove one set of outer brackets. Also, just curious if there is a reason for using a named query in this instance?

Reply:
This is also the query which will need to utilize the `nearObject` [search](https://weaviate.io/developers/weaviate/api/graphql/search-operators#nearobject), using a [`beacon`](https://weaviate.io/developers/weaviate/more-resources/glossary) as the identifying argument for the user (using just the id doesn't work across collections/classes in our experience). So the query will look something like this:

```
{
    Get {
         DiscoveryArtworks(
	    limit: 5
	    nearObject: {
	    beacon: "weaviate://weaviate.stg.artsy.systems/InfiniteDiscoveryUsers/${userID}"
	    }
	    ) {
	        title
	    }
	}
}
```

Reply:
Yeah, that makes sense, do we need even the beacon?

Reply:
I think we can use just an UUID

![image](https://github.com/user-attachments/assets/d220fec7-ae14-4e61-8692-95b1f270578b)

Reply:
When I last tested this, `UUID` only worked when a `nearObject` search used a search argument within the same class/collection as the class/collection being searched. So if I was using a `InfiniteDiscoveryArtwork` object as the search argument, this would work. But because we are using a object from a different class/collection (`InfiniteDiscoveryUsers`) as the search argument to search `InfiniteDiscoveryArtworks` a beacon is needed. I never found good documentation on this, but it was the only way I was previously able to get it to work.